### ISSUE TITLE
Uncaught Error: Cannot decorate a destroyed marker

### ISSUE THREAD

**Issue by despairblue** [steelbrain/linter#1054]
Opened a file in two panes, closed on of them saved and that happen. Don't know how to reproduce it reliably. Will update when I know more.

Linter in question is the `linter-elm-make`.

**Atom Version**: 1.5.0-beta0
**System**: Mac OS X 10.10.5
**Thrown From**: [linter](https://github.com/atom-community/linter) package, v1.11.3

### Stack Trace

Uncaught Error: Cannot decorate a destroyed marker

```
At /Applications/Atom Beta.app/Contents/Resources/app.asar/src/display-buffer.js:1052

Error: Cannot decorate a destroyed marker
  at DisplayBuffer.module.exports.DisplayBuffer.decorateMarker (/Applications/Atom Beta.app/Contents/Resources/app.asar/src/display-buffer.js:1052:15)
  at TextEditor.module.exports.TextEditor.decorateMarker (/Applications/Atom Beta.app/Contents/Resources/app.asar/src/text-editor.js:1316:33)
  at LinterViews.renderBubble [as _renderBubble] (/Users/kietsy/.atom/packages/linter/lib/linter-views.js:122:45)
  at /Users/kietsy/.atom/packages/linter/lib/helpers.coffee:23:18

```

### Commands

```
  2x -1:28.8.0 find-and-replace:select-next (atom-text-editor.editor.is-focused)
     -1:28.3.0 editor:toggle-line-comments (atom-text-editor.editor.is-focused)
  3x -1:27.9.0 core:save (atom-text-editor.editor.is-focused)
     -1:17.8.0 maximize-panes:maximize (atom-text-editor.editor.is-focused)
  3x -1:16.5.0 core:save (atom-text-editor.editor.is-focused)
     -0:56.7.0 core:close (atom-text-editor.editor.is-focused)
     -0:53.4.0 maximize-panes:maximize (atom-text-editor.editor.is-focused)
     -0:44.4.0 core:save (atom-text-editor.editor.is-focused)
```

### Config

```json
{
  "core": {
    "autoHideMenuBar": true,
    "disabledPackages": [
      "atom-color-highlight",
      "atom-pair",
      "atomatigit",
      "es-navigation",
      "filetype-color",
      "grunt-runner",
      "incremental-search",
      "linter-scspell",
      "markdown-scroll-sync",
      "metrics",
      "minimap-color-highlight",
      "minimap-find-and-replace",
      "minimap-git-diff",
      "minimap-highlight-selected",
      "minimap",
      "navigate",
      "starred-package-installer",
      "symbols-tree-view",
      "tab-smart-sort",
      "Tern",
      "test-status",
      "themed-settings",
      "travis-ci-status",
      "tree-view-breadcrumb",
      "tree-view-open-files",
      "welcome",
      "nuclide-click-to-symbol",
      "nuclide-quick-open",
      "linter-jshint",
      "nuclide-file-tree",
      "nuclide-flow",
      "nuclide-file-tree-deux",
      "markdown-preview-opener",
      "nuclide-diagnostics-store",
      "nuclide-diagnostics-ui",
      "nuclide-debugger-node",
      "linter-htmlhint",
      "tree-view"
    ],
    "excludeVcsIgnoredPaths": false,
    "projectHome": "~/vcs/git/"
  },
  "linter": {
    "errorPanelHeight": 70,
    "lintOnChange": true,
    "lintOnFly": false,
    "showErrorPanel": false
  }
}
```

### Installed Packages

```coffee
# User
angularjs, v0.3.4
atom-alignment, v0.12.1
atom-beautify, v0.28.21
atom-ternjs, v0.13.2
atom-zeal, v0.1.3
autocomplete-elixir, v1.5.12
autocomplete-emojis, v2.2.2
autocomplete-paths, v1.0.2
block-travel, v1.0.4
bug-report, v0.7.1
clipboard-plus, v0.5.1
color-picker, v2.1.0
compare-files, v0.6.2
dash, v1.6.0
docblockr, v0.7.3
editorconfig, v1.2.4
file-icons, v1.6.14
git-log, v0.4.1
go-plus, v3.5.2
grammar-token-limit, v0.1.1
haskell-grammar, v0.4.0
highlight-selected, v0.11.1
imdone-atom, v1.3.23
indentation-indicator, v0.6.0
javascript-snippets, v1.2.0
json-schema, v0.1.15
language-babel, v2.12.1
language-docker, v1.1.6
language-dot, v1.6.0
language-elixir, v0.8.0
language-elm, v1.3.2
language-fish-shell, v1.0.2
language-haskell, v1.5.2
language-ini, v1.14.0
language-latex, v0.6.1
language-nimrod, v0.1.0
language-ocaml, v1.1.2
language-puppet, v0.15.0
language-thrift, v1.0.2
language-tss, v0.4.0
latex, v0.28.2
linter, v1.11.3
linter-coffeelint, v1.1.2
linter-elm-make, v0.4.0
linter-eslint, v5.2.7
linter-flow, v4.1.0
linter-js-standard, v3.2.1
linter-js-yaml, v1.2.5
linter-puppet-lint, v0.6.2
linter-shellcheck, v1.1.0
markdown-writer, v2.2.0
maximize-panes, v0.2.0
nuclide, v0.111.0
one-tab, v0.8.0
package-cop, v0.2.10
package-sync, v1.0.0
pdf-view, v0.39.0
pigments, v0.20.0
react, v0.14.1
regex-railroad-diagram, v0.11.0
standard-formatter, v2.0.0
swap-panes, v0.2.0
tabs-to-spaces, v1.0.1
tool-bar, v0.1.11

# Dev
language-elm, v1.3.0
```

**Comment by steelbrain** [comment]
Already fixed in https://github.com/atom-community/linter/pull/995

**Comment by despairblue** [comment]
awesome :clap:

**Comment by brutalcrozt** [comment]
i hit this when editing `.eslintrc.js` which extending other configuration

```
**Atom Version**: 1.5.0
**System**: Microsoft Windows 8 Enterprise
**Thrown From**: [linter](https://github.com/atom-community/linter) package, v1.11.3
```
### Stack Trace
Uncaught Error: Cannot decorate a destroyed marker
```
At C:\Users\enter\AppData\Local\atom\app-1.5.0\resources\app.asar\src\display-buffer.js:1052

Error: Cannot decorate a destroyed marker
    at DisplayBuffer.module.exports.DisplayBuffer.decorateMarker (C:\Users\enter\AppData\Local\atom\app-1.5.0\resources\app.asar\src\display-buffer.js:1052:15)
    at TextEditor.module.exports.TextEditor.decorateMarker (C:\Users\enter\AppData\Local\atom\app-1.5.0\resources\app.asar\src\text-editor.js:1316:33)
    at LinterViews.renderBubble [as _renderBubble] (C:/Users/enter/.atom/packages/linter/lib/linter-views.js:86:43)
    at file:///C:/Users/enter/.atom/packages/linter/lib/helpers.coffee:23:18
```

### Commands

```
     -4:18.5.0 snippets:previous-tab-stop (atom-text-editor.editor.is-focused)
     -4:18.5.0 editor:outdent-selected-rows (atom-text-editor.editor.is-focused)
     -4:12.3.0 core:save (atom-text-editor.editor.is-focused)
     -3:47.1.0 editor:newline (atom-text-editor.editor.is-focused)
  3x -3:40.7.0 editor:delete-to-beginning-of-word (atom-text-editor.editor.is-focused)
     -3:29.9.0 core:backspace (atom-text-editor.editor.is-focused)
     -3:25.5.0 core:save (atom-text-editor.editor.is-focused)
     -0:06.3.0 editor:toggle-line-comments (atom-text-editor.editor.is-focused)
     -0:05.7.0 core:save (atom-text-editor.editor.is-focused)
```

### Config

```json
{
  "core": {
    "audioBeep": false,
    "openEmptyEditorOnStart": false,
    "projectHome": "E:\\Program Files\\xampp\\htdocs\\AA",
    "themes": [
      "atom-dark-ui",
      "base16-tomorrow-dark-theme"
    ]
  },
  "linter": {
    "gutterPosition": "Left",
    "lintOnFlyInterval": 1000
  }
}
```

### Installed Packages

```coffee
# User
angularjs, v0.3.4
angularjs-styleguide-snippets, v0.7.3
docs-snippets, v0.8.0
imdone-atom, v1.3.25
linter, v1.11.3
linter-eslint, v6.0.0
project-manager, v2.7.6

# Dev
No dev packages

**Comment by steelbrain** [comment]
Sorry about this guys, but we're short on a designer here, so it's taking longer than expected

**Comment by agustifernandez** [comment]
How could a designer help then?

**Comment by Arcanemagus** [comment]
@agustifernandez #995 fixes this, but it is waiting on a designer to help with a few issues in https://github.com/AtomLinter/linter-ui-default/.

**Comment by preco21** [comment]
[Enter steps to reproduce below:]

1. ...
2. ...

**Atom Version**: 1.7.0-beta4
**System**: Windows 10
**Thrown From**: [linter](https://github.com/atom-community/linter) package, v1.11.4


### Stack Trace

Uncaught Error: Cannot decorate a destroyed marker

```
At C:\Users\Preco\AppData\Local\atom\app-1.7.0-beta4\resources\app.asar\src\display-buffer.js:1056

Error: Cannot decorate a destroyed marker
    at DisplayBuffer.module.exports.DisplayBuffer.decorateMarker (C:\Users\Preco\AppData\Local\atom\app-1.7.0-beta4\resources\app.asar\src\display-buffer.js:1056:15)
    at TextEditor.module.exports.TextEditor.decorateMarker (C:\Users\Preco\AppData\Local\atom\app-1.7.0-beta4\resources\app.asar\src\text-editor.js:1346:33)
    at LinterViews.renderBubble [as _renderBubble] (C:/Users/Preco/.atom/packages/linter/lib/linter-views.js:86:43)
    at C:\Users\Preco\.atom\packages\linter\lib\helpers.coffee:23:18
```

### Commands

```
     -9:21.2.0 core:copy (atom-text-editor.editor.power-mode.is-focused)
     -9:11.8.0 core:paste (atom-text-editor.editor.power-mode.is-focused)
     -9:11.3.0 core:undo (atom-text-editor.editor.power-mode.is-focused)
     -7:24.5.0 core:move-left (atom-text-editor.editor.power-mode.is-focused)
     -5:14.7.0 window:toggle-dev-tools (atom-text-editor.editor.power-mode.is-focused)
     -3:16.1.0 pane:close (atom-text-editor.editor.power-mode.is-focused)
     -3:13.5.0 core:close (atom-text-editor.editor.power-mode.is-focused)
     -3:11.3.0 pane:split-right-and-copy-active-item (atom-text-editor.editor.power-mode.is-focused)
  2x -3:10.4.0 core:move-down (atom-text-editor.editor.power-mode.is-focused)
     -3:09.9.0 core:close (atom-text-editor.editor.power-mode.is-focused)
     -0:04.0 pane:split-right-and-copy-active-item (atom-text-editor.editor.power-mode.is-focused)
     -0:03.4.0 core:close (atom-text-editor.editor.power-mode.is-focused)
```

### Config

```json
{
  "core": {
    "allowPendingPaneItems": false,
    "autoHideMenuBar": true,
    "disabledPackages": [
      "run-command",
      "language-javascript-better",
      "linter-jshint",
      "linter-jscs",
      "gulp-manager",
      "rainbow-selection",
      "linter-htmlhint"
    ],
    "excludeVcsIgnoredPaths": false,
    "openEmptyEditorOnStart": false,
    "packagesWithKeymapsDisabled": [
      "highlight-line",
      "turbo-javascript",
      "emmet",
      "activate-power-mode",
      "keyboard-scroll"
    ],
    "projectHome": "C:\\Workspace",
    "themes": [
      "atom-material-ui",
      "atom-material-syntax-dark"
    ]
  },
  "linter": {
    "errorPanelHeight": 192,
    "lintOnChangeInterval": 300,
    "subtleLinterErrors": [
      "linter"
    ]
  }
}
```

### Installed Packages

```coffee
# User
activate-power-mode, v0.4.1
atom-bracket-highlight, v0.3.1
atom-material-syntax-dark, v0.2.4
atom-material-ui, v1.2.10
autoclose-html, v0.23.0
autocomplete-html-entities, v0.1.0
autocomplete-modules, v1.4.1
autocomplete-paths, v1.0.2
color-picker, v2.1.1
editorconfig, v1.2.4
emmet, v2.4.3
file-icons, v1.6.20
glowing-cursor, v0.1.1
gutter-shadow, v0.6.0
html-entities, v0.4.0
indentation-indicator, v0.6.0
javascript-snippets, v1.2.1
keyboard-scroll, v0.4.0
last-cursor-position, v0.9.0
linter, v1.11.4
linter-csslint, v1.3.2
linter-eslint, v7.1.3
markdown-scroll-sync, v2.1.2
markdown-themeable-pdf, v1.0.1
minimap, v4.21.0
minimap-autohide, v0.10.1
minimap-cursorline, v0.1.0
minimap-find-and-replace, v4.5.0
minimap-git-diff, v4.2.0
minimap-linter, v1.1.1
open-recent, v5.0.0
pdf-view, v0.50.0
pigments, v0.26.0
react, v0.15.0
swackets, v0.10.0
sync-settings, v0.7.2

# Dev
No dev packages
```

Same issue, which causes when I close a split view.

**Comment by lebowski89** [comment]
Regularly getting this error when having two or more panes open.

**Atom Version**: 1.7.3
**System**: Microsoft Windows 7 Professional 
**Thrown From**: [linter](https://github.com/atom-community/linter) package, v1.11.4


### Stack Trace

Uncaught Error: Cannot decorate a destroyed marker

```
At C:\Users\mike.geldard\AppData\Local\atom\app-1.7.3\resources\app.asar\src\display-buffer.js:1056

Error: Cannot decorate a destroyed marker
    at DisplayBuffer.module.exports.DisplayBuffer.decorateMarker (C:\Users\mike.geldard\AppData\Local\atom\app-1.7.3\resources\app.asar\src\display-buffer.js:1056:15)
    at TextEditor.module.exports.TextEditor.decorateMarker (C:\Users\mike.geldard\AppData\Local\atom\app-1.7.3\resources\app.asar\src\text-editor.js:1346:33)
    at LinterViews.renderBubble [as _renderBubble] (C:/Users/mike.geldard/.atom/packages/linter/lib/linter-views.js:86:43)
    at file:///C:/Users/mike.geldard/.atom/packages/linter/lib/helpers.coffee:23:18
```

### Commands

```
     -3:37.3.0 editor:newline (atom-text-editor.editor.is-focused)
     -3:32.6.0 core:undo (atom-text-editor.editor.is-focused)
     -1:24.8.0 core:delete (atom-text-editor.editor.is-focused)
     -1:23.9.0 core:save (atom-text-editor.editor.is-focused)
```

### Config

```json
{
  "core": {
    "disabledPackages": [
      "Zen",
      "atom-jshint",
      "autocomplete-paths",
      "git-difftool",
      "minimap-git-diff",
      "webbox-color",
      "close-tags",
      "clipboard-plus",
      "wrap-guide",
      "wrap-guide",
      "git-diff-popup",
      "git-diff-details",
      "js-hyperclick",
      "hyperclick"
    ],
    "ignoredNames": [
      ".git",
      ".hg",
      ".svn",
      ".DS_Store",
      "._*",
      "Thumbs.db",
      "node_modules",
      "dist",
      "lib"
    ],
    "openEmptyEditorOnStart": false,
    "packagesWithKeymapsDisabled": [
      "symbol-gen",
      "block-comment",
      "git-diff-details",
      "hyperclick"
    ]
  },
  "linter": {
    "displayLinterInfo": false,
    "displayLinterStatus": false,
    "gutterEnabled": false,
    "showErrorPanel": false,
    "showErrorTabFile": false,
    "showErrorTabProject": false
  }
}
```

### Installed Packages

```coffee
# User
file-icons, v1.7.11
highlight-line, v0.11.1
highlight-selected, v0.11.2
language-babel, v2.20.8
linter, v1.11.4
linter-eslint, v7.2.2
minimap, v4.23.5
minimap-find-and-replace, v4.5.1
minimap-linter, v1.1.1
multi-wrap-guide, v0.23.5
pigments, v0.26.3
react, v0.15.0
sort-lines, v0.14.0
sync-settings, v0.7.2
tabs-to-spaces, v1.0.2
tool-bar, v0.4.0
tool-bar-main, v0.0.9

# Dev
No dev packages
```

**Comment by mnzt** [comment]
@lebowski89 nope, happens in go files, too

**Comment by viddo** [comment]
> This appears to be an ES6 problem...

I can also confirm that it happens in "normal" (non-ES6) JS source files too.

I did something similar as the original reporter, had two panes (vertical split) of same file, edit in one of them that I later closed which seemed to provoke the exceptions.

**Comment by mnzt** [comment]
@viddo I always have two vertical panes open when encountering this issue  so it seems hp be plausible that it's related

**Comment by steelbrain** [comment]
Fixed in v1.11.5
